interfaces/seccomp/backend_test.go

```diff
@@ -0,0 +1,257 @@
+	s.repo = interfaces.NewRepository()
+	s.iface = &interfaces.TestInterface{InterfaceName: "iface"}
+	err = s.repo.AddInterface(s.iface)
+	c.Assert(err, IsNil)
+}
+
+func (s *backendSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+// Tests for Configure() and Deconfigure()
+const sambaYamlV1 = `
+name: samba
+version: 1
+developer: acme
+apps:
+    smbd:
+slots:
+    iface:
+`
+const sambaYamlV1WithNmbd = `
+name: samba
+version: 1
+developer: acme
+apps:
+    smbd:
+    nmbd:
+slots:
+    iface:
+`
+const sambaYamlV2 = `
+name: samba
+version: 2
+developer: acme
+apps:
+    smbd:
+slots:
+    iface:
+`
+
+func (s *backendSuite) TestInstallingSnapWritesProfiles(c *C) {
+	developerMode := false
+	s.installSnap(c, developerMode, sambaYamlV1)
+	profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+	// file called "snap.sambda.smbd" was created
+	_, err := os.Stat(profile)
+	c.Check(err, IsNil)
+}
+
+func (s *backendSuite) TestRemovingSnapRemovesProfiles(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1)
+		s.removeSnap(c, snapInfo)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+		// file called "snap.sambda.smbd" was removed
+		_, err := os.Stat(profile)
+		c.Check(os.IsNotExist(err), Equals, true)
+	}
+}
+
+func (s *backendSuite) TestUpdatingSnapToOneWithMoreApps(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1)
+		snapInfo = s.updateSnap(c, snapInfo, developerMode, sambaYamlV1WithNmbd)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.nmbd")
+		// file called "snap.sambda.nmbd" was created
+		_, err := os.Stat(profile)
+		c.Check(err, IsNil)
+		s.removeSnap(c, snapInfo)
+	}
+}
+
+func (s *backendSuite) TestUpdatingSnapToOneWithFewerApps(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1WithNmbd)
+		snapInfo = s.updateSnap(c, snapInfo, developerMode, sambaYamlV1)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.nmbd")
+		// file called "snap.sambda.nmbd" was removed
+		_, err := os.Stat(profile)
+		c.Check(os.IsNotExist(err), Equals, true)
+		s.removeSnap(c, snapInfo)
+	}
+}
+
+func (s *backendSuite) TestRealDefaultTemplateIsNormallyUsed(c *C) {
+	snapInfo, err := snap.InfoFromSnapYaml([]byte(sambaYamlV1))
+	c.Assert(err, IsNil)
+	// NOTE: we don't call seccomp.MockTemplate()
+	err = s.backend.Configure(snapInfo, false, s.repo)
+	c.Assert(err, IsNil)
+	profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+	data, err := ioutil.ReadFile(profile)
+	c.Assert(err, IsNil)
+	for _, line := range []string{
+		// NOTE: a few randomly picked lines from the real profile.  Comments
+		// and empty lines are avoided as those can be discarded in the future.
+		"deny kexec_load\n",
+		"deny create_module\n",
+		"symlink\n",
+		"socket\n",
```

Comment:
I wanted a preprocessor in the launcher for other reasons and have implemented that just now. I'll be adding '@deny' rules so instead of 'deny syscall' I'm going to use '@deny syscall' so that 'deny' doesn't collide with potential future syscalls. In other words, I'll handle the launcher. We can do 's/^deny/@deny/g' in template.go and builtin/ in a future commit.